docs/source/user_guide/get_started/examples/pw_tutorial.rst

```diff
@@ -51,63 +61,126 @@ This is the input file of Quantum Espresso that we will try to execute. It consi
+  You can use directly the ``.get()`` method, for instance::
 
-    code = Code.get(label='pw-5.1', machinename='TheHive')
+    code = Code.get(label='pw-6.3', machinename='TheHive')
 
-  or even more generally get the code from its (integer) PK::
+  or even more generally get the code from its (integer) ``PK``::
 
     code = load_node(PK)
 
+Once you have a code, you can start to assemble the inputs to run
+a PWscf calculation.
+
+.. note:: To learn more about calculations and processes in AiiDA you can
+  refer to the :ref:`working_calculations` and :ref:`concepts_processes` sections 
+  of the AiiDA manual
+  Remember that what is shown here refers to the Quantum ESPRESSO plugin: 
+  different codes will in general required different inputs.
+
+
+Preparing a calculation
+------------------------ 
+
+To make it easier to prepare the inputs of an AiiDA calculation, process classes 
+have a ``get_builder`` method, that simplify the access to the inputs of a calculation.
+
+To use it, you can load the calculation class that you need through the 
+``CalculationFactory`` ::
+ 
+  PwCalculation = CalculationFactory('quantumespresso.pw')
+  builder = PwCalculation.get_builder()
+
+In a similar way, you can use the ``get_builder`` utility from a code, with the advantage 
+that the *code* input gets automatically populated: ::
+
+  builder = code.get_builder()
+
+
+If you are using the builder to a verdi shell, you will 
+see all the inputs that are available for a calculation by pressing the ``TAB`` key 
+after typing ``builder.``. To understand what type of data is expected for a 
+particular input, you can append the ``?`` or the ``??`` at the end of an input. For example: ::
+
+    >>> builder.structure?
+    Type:        property
+    String form: <property object at 0x7f58bdcc6728>
+    Docstring:   {"name": "structure", "required": "True", "valid_type": "<class '`aiida.orm.nodes.data.structure.StructureData`'>", "help": "The input structure."}
+
+In this case, the helper lets you know you what kind of data it expects 
+(:py:class:`StructureData <aiida.orm.nodes.data.structure.StructureData>`),
+that it is a *required* input for the calculation, and a short string explaining what the *structure* input represents.
+
+The plugin requires at least the presence of:
+
+    - An input structure;
+    - A k-points mesh, in the form of a :py:class:`KpointsData <aiida.orm.nodes.data.array.kpoints.KpointsData>` object;
+    - Pseudopotential files for the atomic species present;
+    - A parameters dictionary, that contains the details of the Quantum ESPRESSO calculation;
+
+
+Other inputs are optional, for example:
+
+    - *metadata* is a :py:class:`Dict <aiida.orm.nodes.data.dict.Dict>` dictionary of inputs that modify slightly the behaviour of a processes;
+    - *options* is a :py:class:`Dict <aiida.orm.nodes.data.dict.Dict>` dictionary that provides access to more advanced, non-default feature of the code.
```

Comment:
Same here: it is *not* a `Dict` node, but just a plain python dictionary
But technically, `options` is just a member of `metadata`, just like `label` and `description` so maybe it does not need a separate line

Reply:
Yes, that was a mistake, I meant *settings* there, that is a separate input (just because it is mentioned in the `gamma only` kpoints section).

I'll just correct the entry and leave the description intact, is that is ok for you